lib/ansible/modules/network/fortios/fortios_webfilter.py

```diff
@@ -0,0 +1,548 @@
+  returned: always
+  type: string
+  sample: "success"
+vdom:
+  description: Virtual domain used
+  returned: always
+  type: string
+  sample: "root"
+version:
+  description: Version of the FortiGate
+  returned: always
+  type: string
+  sample: "v5.6.3"
+
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+fos = None
+
+
+def login(data):
+    host = data['host']
+    username = data['username']
+    password = data['password']
+
+    fos.debug('on')
+    fos.https('off')
+
+    fos.login(host, username, password)
+
+
+def filter_wf_url_data(json):
+    option_list = ['id', 'name', 'comment',
+                   'one-arm-ips-urlfilter',
+                   'ip-addr-block', 'entries']
+    dictionary = {}
+
+    for attribute in option_list:
+        if attribute in json:
+            dictionary[attribute] = json[attribute]
+
+    return dictionary
+
+
+def filter_wf_content_data(json):
+    option_list = ['id', 'name', 'comment',
+                   'entries']
+    dictionary = {}
+
+    for attribute in option_list:
+        if attribute in json:
+            dictionary[attribute] = json[attribute]
+
+    return dictionary
+
+
+def webfilter_url(data, fos):
+    vdom = data['vdom']
+    wf_url_data = data['webfilter_url']
+    url_data = filter_wf_url_data(wf_url_data)
+
+    if wf_url_data['state'] == "present":
+        return fos.set('webfilter',
+                       'urlfilter',
+                       data=url_data,
+                       vdom=vdom)
+
+    elif wf_url_data['state'] == "absent":
+        return fos.delete('webfilter',
+                          'urlfilter',
+                          mkey=url_data['id'],
+                          vdom=vdom)
+
+
+def webfilter_content(data, fos):
+    vdom = data['vdom']
+    wf_content_data = data['webfilter_content']
+    content_data = filter_wf_content_data(wf_content_data)
+
+    if wf_content_data['state'] == "present":
+        return fos.set('webfilter',
+                       'content',
+                       data=content_data,
+                       vdom=vdom)
+
+    elif wf_content_data['state'] == "absent":
+        return fos.delete('webfilter',
+                          'content',
+                          mkey=content_data['id'],
+                          vdom=vdom)
+
+
+def fortios_webfilter(data, fos):
+    host = data['host']
+    username = data['username']
+    password = data['password']
+    fos.https('off')
+    fos.login(host, username, password)
+
+    methodlist = ['webfilter_url', 'webfilter_content', 'webfilter_profile']
+    for method in methodlist:
+        if data[method]:
+            resp = eval(method)(data, fos)
+            break
+
+    fos.logout()
+    return not resp['status'] == "success", resp['status'] == "success", resp
+
+
+def main():
+    fields = {
+        "host": {"required": True, "type": "str"},
+        "username": {"required": True, "type": "str"},
+        "password": {"required": False, "type": "str"},
```

Comment:
Need no_log here,
see https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/module_utils/basic.rst

This will also need adding to any options that have secrets